scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -468,51 +466,28 @@ protected NodeSet getRMNodes(Map<JobId, JobDescriptor> jobMap, int neededResourc
         }
     }
 
+    private void updateVariablesForTasksToSchedule(Map<JobId, JobDescriptor> jobMap, LinkedList<EligibleTaskDescriptor> tasksToSchedule) throws IOException, ClassNotFoundException {
+        for (EligibleTaskDescriptor taskDescriptor : tasksToSchedule) {
+            InternalJob associatedJob = jobMap.get(taskDescriptor.getJobId()).getInternal();
+            InternalTask internalTask = associatedJob.getIHMTasks().get(taskDescriptor.getTaskId());
+            internalTask.updateVariables(schedulingService);
+        }
+    }
+
     /**
      * Create bindings which will be used by selection scripts for the given tasks
      */
     private Map<String, Serializable> createBindingsForSelectionScripts(InternalJob job, EligibleTaskDescriptor etd, InternalTask task) throws IOException, ClassNotFoundException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "etd". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1172)